private/pkg/protosource/protosource.go

```diff
@@ -841,6 +841,44 @@ func PackageToNameToService(files ...File) (map[string]map[string]Service, error
 	return packageToNameToService, nil
 }
 
+// PackageToDirectlyImportedPackageToFileImports maps packages to directly imported packages
+// to the FileImports that import this package.
+//
+// For example, if package a imports package b via c/d.proto and c/e.proto, this will have
+// a -> b -> [c/d.proto, c/e.proto].
+//
+// A directly imported package will not be equal to the package, i.e. there will be no a -> a.
+//
+// Files with no packages are included with key "" to be consistent with other functions.
+func PackageToDirectlyImportedPackageToFileImports(files ...File) (map[string]map[string][]FileImport, error) {
```

Comment:
This one's a mouthful, but I can't think of anything better 🤦 

We _could_ do something like

```
type ImportGraph map[string]map[string][]FileImport
```

Then this would be renamed `PackageToImportGraph`, but that's probably more trouble than it's worth.

Reply:
We don't do that anywhere else in the codebase though - I'm super anti-creating types like this

Reply:
Yea we can skip it, we only call this function internally anyway so it's fine as-is.